### ISSUE TITLE
[flang] ERROR STOP default code is wrong

### ISSUE THREAD

**Issue by sscalpone** [llvm/llvm-project#66581]
Looks like a problem lowering the default value of the exit code.

```
% flang-new estop.f90
% a.out
Fortran ERROR STOP
% echo $?
0
```

The exit code ought to be 1.

> If the stop-code in an ERROR STOP statement is of type character or does not appear, it is recommended that a processor-dependent nonzero value be supplied as the process exit status, if the processor supports that concept.

**Comment by aidenfoxivey** [comment]
Would love to give this a shot. Would that be okay?

**Comment by jeanPerier** [comment]
> Would love to give this a shot. Would that be okay?

Yes @aidenfoxivey, thanks! Please assign the issue to yourself and go for it.

**Comment by aidenfoxivey** [comment]
> > Would love to give this a shot. Would that be okay?
> 
> Yes @aidenfoxivey, thanks! Please assign the issue to yourself and go for it.

I'm actually a first time contributor. How do you assign yourself to different tasks?

**Comment by aidenfoxivey** [comment]
> > > > Would love to give this a shot. Would that be okay?
> > > 
> > > 
> > > Yes @aidenfoxivey, thanks! Please assign the issue to yourself and go for it.
> > 
> > 
> > I'm actually a first time contributor. How do you assign yourself to different tasks?
> 
> The github way, there is an "Assignees" menu on the top right of this issue, if you click on it you should be able to assign yourself. Maybe you do not have the permission to do it though, so I just did it for you.

Awesome, thank you. 😊

**Comment by aidenfoxivey** [comment]
@jeanPerier My apologies to ask for assistance on this. (Also not sure this is the right place to ask for help on this.) 

I can't seem to get `flang-new` to build on macOS 14.0. I followed [this guide](https://github.com/llvm/llvm-project/blob/main/flang/docs/Overview.md) in the docs and the build has given me a `flang-new` compiler that returns 
```
ld: library 'System' not found
flang-new: error: linker command failed with exit code 1 (use -v to see invocation)
```
 when I run it on a [program like this one](https://www.mrao.cam.ac.uk/~pa/f90Notes/HTMLNotesnode40.html).

Is this maybe something worth opening its own issue about or am I making some sort of mistake?

Thank you so much for your time. I'm trying to get into working on LLVM and any help is dearly appreciated.

**Comment by aidenfoxivey** [comment]
Whoops, nevermind. Got it! Sorry to waste any time.

Would it be worth a mention in the documentation that you need to export `LIBRARY_PATH` with the path `/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib` on macOS to use it?

Not sure if that's outside the scope of the issue.

**Comment by psteinfeld** [comment]
> Whoops, nevermind. Got it! Sorry to waste any time.
> 
> Would it be worth a mention in the documentation that you need to export `LIBRARY_PATH` with the path `/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib` on macOS to use it?
> 
> Not sure if that's outside the scope of the issue.

@aidenfoxivey, it would be great to add instructions on how to build on macOS.  I have a MacBook Pro, and I'd love to do development on it.  But that's a different topic from the STOP code, so it should be put into a separate update.

**Comment by psteinfeld** [comment]
> @aidenfoxivey, it would be great to add instructions on how to build on macOS. I have a MacBook Pro, and I'd love to do development on it. But that's a different topic from the STOP code, so it should be put into a separate update.

Also, please make me a reviewer if you do add instructions for building on macOS.

**Comment by megh-bari** [comment]
Is this issue open . I would like to give it a try

**Comment by clementval** [comment]
> Is this issue open . I would like to give it a try

If you have a fix please send a PR.

**Comment by megh-bari** [comment]
Where should I find this code?
